engine/src/main/java/com/sandboni/core/engine/sta/operation/ReflectionCallTestsOperation.java

```diff
@@ -0,0 +1,46 @@
+package com.sandboni.core.engine.sta.operation;
+
+import com.sandboni.core.engine.sta.graph.Edge;
+import com.sandboni.core.engine.sta.graph.vertex.TestVertex;
+import com.sandboni.core.engine.sta.graph.vertex.Vertex;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.interfaces.ShortestPathAlgorithm;
+import org.jgrapht.alg.shortestpath.BellmanFordShortestPath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import static com.sandboni.core.engine.common.StreamHelper.emptyIfFalse;
+import static com.sandboni.core.engine.sta.graph.vertex.VertexInitTypes.REFLECTION_CALL_VERTEX;
+import static com.sandboni.core.engine.sta.graph.vertex.VertexInitTypes.START_VERTEX;
+
+public class ReflectionCallTestsOperation extends AbstractGraphOperation<SetResult<TestVertex>> {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReflectionCallTestsOperation.class);
+
+    private final Set<TestVertex> allTests;
+
+    ReflectionCallTestsOperation(Set<TestVertex> allTests) {
+        Objects.requireNonNull(allTests, INVALID_ARGUMENT);
+        this.allTests = new HashSet<>(allTests);
+    }
+
+    @Override
+    public SetResult<TestVertex> execute(Graph<Vertex, Edge> graph) {
+        ShortestPathAlgorithm<Vertex, Edge> algorithm = new BellmanFordShortestPath<>(graph);
+        return new SetResult<>(emptyIfFalse(graph.containsVertex(START_VERTEX) && graph.containsVertex(REFLECTION_CALL_VERTEX),
```

Comment:
See if there's a way to avoid looking for paths again, we are already doing it in Related Tests